shared/javascripts/context_messenger.js

```diff
@@ -471,12 +471,39 @@ if (Privly === undefined) {
   SafariAdapter.prototype.setListener = function (callback) {
     if (this.getContextName() === 'BACKGROUND_SCRIPT') {
       safari.application.addEventListener("message", function(payload) {
-        callback(payload);
+        if (payload.name !== "undefined" && payload.name === "privlyMessage") {
+          callback(payload.message);
+          payload = payload.message;
+        } else {
+          callback(payload);
+        }
+        // Update the options/changed values in the background page
+        if (payload.body !== "undefined" && payload.body.action === "options/changed") {
```

Comment:
I have updated the PR to have only the initial commit now. If you think the code looks good, then the PR can be merged.